x-pack/plugins/upgrade_assistant/server/lib/apm/index.ts

```diff
@@ -326,6 +339,16 @@ export const apmReindexScript = `
           }
       }
 
+      // context.page -> error.page,transaction.page
+      def page = context.remove("page");
+      if (page != null) {
+          if (ctx._source.processor.event == "transaction") {
+              ctx._source.transaction.page = page;
```

Comment:
Do we need to protect against `_source.transaction` not existing?

Reply:
When the event is a `transaction` it should always have a `_source.transaction` field. Same assumption holds for `error` documents. We aimed to introduce guards where it makes sense for valid data to be nil.